client/lcd/root_test.go

```diff
@@ -0,0 +1,45 @@
+package lcd
+
+import (
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/codec"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSwaggerAsRootRoute(t *testing.T) {
+	cmd := newTestCmd()
+	go cmd.Execute()
+
+	// Wait until is up
+	time.Sleep(time.Millisecond * 200)
+
+	res, err := http.Get("http://127.0.0.1:1317/")
```

Comment:
where is the server being setup? i may be looking at this wrong or the node with the rest server is being set somewhere else?

Reply:
in theory this launches a server in localhost, I just execute the command related to rest. 

It is executed in line 21. That command launches a server and I set it up with the flag

'''
	viper.Set(flags.FlagListenAddr, "tcp://localhost:1317")
'''

Try to execute this test in your local machine if you want. Should pass, let me know if not. Then we need to discover a better way to make it pass in CI! :D